catalog/ui/catalog-ui-search/src/main/webapp/js/cql.js

```diff
@@ -72,26 +73,28 @@ define([
         },
 
         follows = {
-            LPAREN: ['NOT', 'GEOMETRY', 'SPATIAL', 'PROPERTY', 'VALUE', 'LPAREN'],
-            RPAREN: ['NOT', 'LOGICAL', 'END', 'RPAREN'],
-            PROPERTY: ['COMPARISON', 'BETWEEN', 'COMMA', 'IS_NULL', 'BEFORE', 'AFTER', 'DURING'],
+            ROOT_NODE: ['NOT', 'GEOMETRY', 'SPATIAL', 'FILTER_FUNCTION', 'PROPERTY', 'LPAREN'],
+            LPAREN: ['NOT', 'GEOMETRY', 'SPATIAL', 'FILTER_FUNCTION', 'PROPERTY', 'VALUE', 'LPAREN', 'RPAREN'],
+            RPAREN: ['NOT', 'LOGICAL', 'END', 'RPAREN', 'COMPARISON', 'COMMA'],
+            PROPERTY: ['COMPARISON', 'BETWEEN', 'COMMA', 'IS_NULL', 'BEFORE', 'AFTER', 'DURING', 'RPAREN'],
             BETWEEN: ['VALUE'],
             IS_NULL: ['END'],
             COMPARISON: ['RELATIVE', 'VALUE', 'BOOLEAN'],
-            COMMA: ['GEOMETRY', 'VALUE', 'UNITS', 'PROPERTY'],
+            COMMA: ['FILTER_FUNCTION', 'GEOMETRY', 'VALUE', 'UNITS', 'PROPERTY'],
             VALUE: ['LOGICAL', 'COMMA', 'RPAREN', 'END'],
             BOOLEAN: ['RPAREN'],
             SPATIAL: ['LPAREN'],
             UNITS: ['RPAREN'],
-            LOGICAL: ['NOT', 'VALUE', 'SPATIAL', 'PROPERTY', 'LPAREN'],
+            LOGICAL: ['FILTER_FUNCTION', 'NOT', 'VALUE', 'SPATIAL', 'PROPERTY', 'LPAREN'],
             NOT: ['PROPERTY', 'LPAREN'],
             GEOMETRY: ['COMMA', 'RPAREN'],
             BEFORE: ['TIME'],
             AFTER: ['TIME'],
             DURING: ['TIME_PERIOD'],
             TIME: ['LOGICAL', 'RPAREN', 'END'],
             TIME_PERIOD: ['LOGICAL', 'RPAREN', 'END'],
-            RELATIVE: ['RPAREN']
+            RELATIVE: ['RPAREN'],
+            FILTER_FUNCTION: ['LPAREN','PROPERTY','VALUE','RPAREN'],
```

Comment:
In what case can we do FILTER_FUNCTION RPAREN?
Should we not have a trailing comma?